File: src/main/java/net/dashflight/data/keys/RSAKeyPairProvider.java
package net.dashflight.data.keys;

import java.security.interfaces.RSAPrivateKey;
import java.security.interfaces.RSAPublicKey;

/**
 * Responsible for returning a valid RSA key pair.
 */
public interface RSAKeyPairProvider {

    RSAPublicKey getPublicKey();

    RSAPrivateKey getPrivateKey();

}

File: src/main/java/net/dashflight/data/keys/RSAKeyPairProvider.kt
package net.dashflight.data.keys

import java.security.interfaces.RSAPrivateKey
import java.security.interfaces.RSAPublicKey

/**
 * Responsible for returning a valid RSA key pair.
 */
interface RSAKeyPairProvider {
    val publicKey: RSAPublicKey?
    val privateKey: RSAPrivateKey?
}